patches/minecraft/net/minecraft/world/WorldServer.java.patch

```diff
@@ -89,7 +89,16 @@
                  {
                      this.field_73005_l = this.field_73005_l * 3 + 1013904223;
                      int i1 = this.field_73005_l >> 2;
-@@ -363,7 +381,7 @@
+@@ -357,13 +375,15 @@
+ 
+                     if (this.func_175727_C(blockpos))
+                     {
+-                        this.func_72942_c(new EntityLightningBolt(this, (double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p()));
++                    	if (!net.minecraftforge.common.ForgeHooks.onLightningStrike(this.field_73011_w.field_76579_a,blockpos)) {
++                    		this.func_72942_c(new EntityLightningBolt(this, (double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p()));
++                    	}
```

Comment:
wouldn't the event want to specify the blockpos to the entity creation possibly? Or you expect the event to create the entity itself and cancel the event?